Doppler.HtmlEditorApi/Domain/TemplateData.cs

```diff
@@ -0,0 +1,28 @@
+namespace Doppler.HtmlEditorApi.Domain;
+
+public abstract record TemplateData(
+    string HtmlCode,
+    string Meta,
+    string PreviewImage,
+    TemplateAccess Access)
+{ }
+
+public sealed record OwnTemplateData(
+    string HtmlCode,
+    string Meta,
+    string PreviewImage)
+    : TemplateData(HtmlCode, Meta, PreviewImage, TemplateAccess.Own);
+
+public sealed record PublicTemplateData(
+    string HtmlCode,
+    string Meta,
+    string PreviewImage)
+    : TemplateData(HtmlCode, Meta, PreviewImage, TemplateAccess.Public);
+
+public sealed record InactiveTemplateData()
+    : TemplateData(null, null, null, TemplateAccess.Inactive);
```

Comment:
In my opinion, own/public or active/inactive are not types.

Reply:
What would you call that?